manticore/core/smtlib/visitors.py

```diff
@@ -488,12 +479,12 @@ def visit_BoolOr(self, expression, a, b):
             return a
 
     def visit_BitVecITE(self, expression, *operands):
```

Comment:
I think *operands are the potentially modified expression.operands.
operand[0] should be something like simplify(expression.operand[0])

Reply:
Thank you for taking a look at this @feliam. I don't quite understand your comment. Is that a suggestion to change the code or a statement about what the code is doing currently?

I can do some more experimentation if you think the changes in this PR aren't completely correct. 

I'm also open to you committing directly to this branch if you have a fix that has everything in trailofbits/manticore-examples#35 passing.

Reply:
I think Felipe's comment is in regards to the current behavior. Visit methods get called recursively, so the `*operands` argument will contain the results of running `visit_<x>` on each of the operands to the BitVecITE. Whereas, if you access the operands via `expression.operands[n]`, you'll get a raw version of  each operand itself instead of the output of the visitor.